From ec6732450459329b99fcc24f55c2316a297b9758 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Fri, 26 Sep 2025 18:06:28 +0200
Subject: [PATCH] Make dialog as modal only for the window

The file chooser dialog is ApplicationModal by default, causing that
a dialog opened for one portal request blocks dialogs requested from other
applications. Setting the dialog as WindowModal avoids this problem.
---
 src/filechooser.cpp    | 4 ++--
 src/filedialoghelper.h | 1 +
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/filechooser.cpp b/src/filechooser.cpp
index da49a3a..35cce6b 100644
--- a/src/filechooser.cpp
+++ b/src/filechooser.cpp
@@ -238,7 +238,7 @@ namespace LXQt
         auto fileDialog = FileDialogHelper::createFileDialogHelper();
         Utils::setParentWindow(&fileDialog->dialog(), parent_window);
         fileDialog->setWindowTitle(title);
-        fileDialog->setModal(modalDialog);
+        fileDialog->setWindowModality(modalDialog ? Qt::WindowModal : Qt::NonModal);
         fileDialog->setFileMode(directory ? QFileDialog::Directory : (multipleFiles ? QFileDialog::ExistingFiles : QFileDialog::ExistingFile));
         if (!acceptLabel.isEmpty())
             fileDialog->setLabelText(QFileDialog::Accept, acceptLabel);
@@ -353,7 +353,7 @@ namespace LXQt
         auto fileDialog = FileDialogHelper::createFileDialogHelper();
         Utils::setParentWindow(&fileDialog->dialog(), parent_window);
         fileDialog->setWindowTitle(title);
-        fileDialog->setModal(modalDialog);
+        fileDialog->setWindowModality(modalDialog ? Qt::WindowModal : Qt::NonModal);
         fileDialog->setFileMode(QFileDialog::AnyFile);
         fileDialog->setAcceptMode(QFileDialog::AcceptSave);
         if (!acceptLabel.isEmpty())
diff --git a/src/filedialoghelper.h b/src/filedialoghelper.h
index f374b2e..4f9129d 100644
--- a/src/filedialoghelper.h
+++ b/src/filedialoghelper.h
@@ -50,6 +50,7 @@ namespace LXQt
         inline void setFileMode(QFileDialog::FileMode mode) { options()->setFileMode(static_cast<QFileDialogOptions::FileMode>(mode)); }
         inline void setWindowTitle(const QString & title) { options()->setWindowTitle(title); }
         inline void setModal(bool modal) { dialog().setModal(modal); }
+        inline void setWindowModality(Qt::WindowModality modality) { dialog().setWindowModality(modality); }
         inline void setLabelText(QFileDialog::DialogLabel label, const QString &text) { options()->setLabelText(static_cast<QFileDialogOptions::DialogLabel>(label), text); };
         inline void setNameFilters(const QStringList &filters) { options()->setNameFilters(filters); }
         inline void setAcceptMode(QFileDialog::AcceptMode mode) { options()->setAcceptMode(static_cast<QFileDialogOptions::AcceptMode>(mode)); }
